spectator/id.py

```diff
@@ -1,13 +1,63 @@
+import logging
+from enum import Enum
 from typing import Dict, Optional
 
 
+class MeterIdError(Enum):
+    NONE = 0
+    NAME_TOO_SHORT = 1
+    NAME_TOO_LONG = 2
+    TAG_KEY_NOT_A_STR = 3
+    TAG_KEY_TOO_SHORT = 4
+    TAG_KEY_TOO_LONG = 5
+    TAG_VALUE_NOT_A_STR = 6
+    TAG_VALUE_TOO_SHORT = 7
+    TAG_VALUE_TOO_LONG = 8
```

Comment:
For most of these, spectatord would report them in its log file based on the response from the server side validation. Only the "not a string" checks would really be needed here. Having the others probably does help a bit with debugging as it would be local to the python usage, but would means this client would need to get updated if those rules change instead of just the server side checks. So it adds to the maintenance burden, especially if replicated across other clients.

Reply:
This is a good point. I had been thinking that while the extra checks provide more direct feedback to developers, I do not want to increase the maintenance burden for these thin clients and would prefer to rely on SpectatorD as a central authority for validation.

We can think of the line of demarcation between the client and the agent as the ability to produce a valid SpectatorD protocol line. The not-a-str check is important to keep, because this prevents the generation of the line entirely, and the existing invalid character substitution is important, because that may or may not produce a broken line, depending on the characters included.